rice-namenode/source/ClientNamenodeProtocolImpl.cc

```diff
@@ -195,56 +191,65 @@ std::string ClientNamenodeTranslator::rename(std::string input) {
 	return Serialize(res);
 }
 
-std::string ClientNamenodeTranslator::rename2(std::string input) {
-	Rename2RequestProto req;
-	req.ParseFromString(input);
-	logMessage(req, "Rename2 ");
-	Rename2ResponseProto res;
-	return Serialize(res);	
-}
-
-std::string ClientNamenodeTranslator::append(std::string input) {
-	AppendRequestProto req;
-	req.ParseFromString(input);
-	logMessage(req, "Append ");
-	AppendResponseProto res;
-	return Serialize(res);
-}
-
-/**
- * This is effectively an append, so we do not support it 
- */
-std::string ClientNamenodeTranslator::concat(std::string input) {
-	ConcatRequestProto req;
-	req.ParseFromString(input);
-	logMessage(req, "Concat ");
-	ConcatResponseProto res;
-	return Serialize(res);	
-}
-
-/**
- * TODO we might support this in the future!
- */ 
 std::string ClientNamenodeTranslator::setPermission(std::string input) {
-	SetPermissionResponseProto res;
-	return Serialize(res);
+		SetPermissionResponseProto res;
+			return Serialize(res);
 }
-
+// ----------------------- COMMANDS WE DO NOT SUPPORT ------------------
 /**
+ * When asked to do an unsupported command, we'll be returning a
+ * method-not-found proto.  The code in question is very similar to
+ * GetErrorRPCHeader, but will actually occur further up - see rpcserver.cc's
+ * method handle_rpc. Whenever (iter != dispatch_table.end()) is false, it
+ * basically means that we couldn't find a corresponding method in this file
+ * here. 
+ *
+ * As such, it will go ahead and create the error header and send it back
+ * along, without ever having to call any methods in this file. So there is
+ * no need to ever worry about methods we just flat out don't support in this 
+ * file.
+ *
+ * That being said, the following is a short list of some common commands we don't
+ * support, and our reasons for not supporting them:
+ *
+ * 1. setReplication:
+ * The actual block replication is not expected to be performed during  
+ * this method call. The blocks will be populated or removed in the 
+ * background as the result of the routine block maintenance procedures.
+ * Basically, cannot set replication to something new.
+ *
+ * 2. append: 
+ * Appends are not supported
+ *
+ * 3. concat:
+ * Effectively an append, so we don't support it
+ *
+ * 4. recoverLease:
  * While we expect clients to renew their lease, we should never allow
  * a client to "recover" a lease, since we only allow a write-once system
- */ 
-std::string ClientNamenodeTranslator::recoverLease(std::string input) {
```

Comment:
We must still suppor this!! It has a boolean which we set to false, which is better than throwing an exception :)